packages/engine/src/interaction/behaviors/interact.ts

```diff
@@ -3,20 +3,40 @@ import { Entity } from "../../ecs/classes/Entity";
 import { Interactive } from "../components/Interactive";
 import { getComponent, hasComponent } from "../../ecs/functions/EntityFunctions";
 import { Interacts } from "../components/Interacts";
+import { LifecycleValue } from "../../common/enums/LifecycleValue";
+import { DefaultInput } from "../../templates/shared/DefaultInput";
+import { Input } from "../../input/components/Input";
 
 /**
  *
  * @param entity the one who interacts
  * @param args
  * @param delta
  */
+
+
+const startedPosition = new Map<Entity,any>();
+
+
 export const interact:Behavior = (entity: Entity, args:any, delta): void => {
   if (!hasComponent(entity, Interacts)) {
     console.error('Attempted to call interact behavior, but actor does not have Interacts component');
     return;
   }
   const { focusedInteractive:focusedEntity } = getComponent(entity, Interacts);
 
+  const mouseScreenPosition = getComponent(entity, Input).data.get(DefaultInput.SCREENXY);
+  if (args.phaze === LifecycleValue.STARTED ){
+    startedPosition.set(entity,mouseScreenPosition.value)
+    
+  }
+  
+  const startedMousePosition = startedPosition.get(entity);
+  
+  // console.log('Mouse position on START',startedMousePosition)
+  // console.log('Current mouse position', mouseScreenPosition.value)
+  
+  if (startedMousePosition == mouseScreenPosition.value){
```

Comment:
i think indentations is wrong inside this block.
may be you should replace
```
if (startedMousePosition == mouseScreenPosition.value){
   ...
   ...
}
```
with
```
if (startedMousePosition !== mouseScreenPosition.value){
   return
}
...
...
```